web/docs/auth/username-and-pass.md

```diff
@@ -1,3 +1,659 @@
+
+## Using Auth 
+
+To read more about how to set up the logout button and how to get access to the logged-in user in our client and server code, read the [using auth docs](/docs/auth/overview).
+
+
+## Customizing the Auth Flow
+
+If you require more control in your authentication flow, you can achieve that in the following ways:
+1. Create your UI and use `signup` and `login` actions.
+1. Create your custom sign-up and login [actions](#) which uses the Prisma client, along with your custom code.
+
+### 1. Using the `signup` and `login` actions
+
+#### `login()`
+An action for logging in the user.
+
+It takes two arguments:
+
+  - `username: string` <Required />
+
+  Username of the user logging in.
+
+  - `password: string` <Required />
+
+  Password of the user logging in.
+
+You can use it like this:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```jsx title="client/pages/auth.jsx"
+// Importing the login action 👇
+import login from '@wasp/auth/login'
+
+import { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { Link } from 'react-router-dom'
+
+export function LoginPage() {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState(null)
+  const history = useHistory()
+
+  async function handleSubmit(event) {
+    event.preventDefault()
+    try {
+      await login(username, password)
+      history.push('/')
+    } catch (error) {
+      setError(error)
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      {/* ... */}
+    </form>
+  );
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```tsx title="client/pages/auth.tsx"
+// Importing the login action 👇
+import login from '@wasp/auth/login'
+
+import { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { Link } from 'react-router-dom'
+
+export function LoginPage() {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [error, setError] = useState<Error | null>(null)
+  const history = useHistory()
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    try {
+      await login(username, password)
+      history.push('/')
+    } catch (error: unknown) {
+      setError(error as Error)
+    }
+  }
+
+  return (
+    <form onSubmit={handleSubmit}>
+      {/* ... */}
+    </form>
+  );
+}
+```
+</TabItem>
+</Tabs>
+
+:::note
+When using the exposed `login()` function, make sure to implement your redirect on success login logic (e.g. redirecting to home).
+:::
+
+#### `signup()`
+An action for signing up the user. This action does not log in the user, you still need to call `login()`.
+
+It takes one argument:
+- `userFields: object` <Required />
+
+  Auth-related fields (either `username` or `email` and `password`) of the user entity which was declared in `auth`.
```

Comment:
It's confusing to mention email here: what happens if you try to enable both email auth and username and password auth? And does `login()` take username or email.

It feels like we shouldn't mention `email` here since this page isn't about email auth.

Reply:
Makes sense, removed the `email` mention 👍